HPXMLtoOpenStudio/resources/hpxml_defaults.rb

```diff
@@ -211,20 +211,26 @@ def self.apply_neighbor_buildings(hpxml)
     end
   end
 
-  def self.apply_building_occupancy(hpxml, nbeds)
+  def self.apply_building_occupancy(hpxml, nbeds, schedules_file)
     if hpxml.building_occupancy.number_of_residents.nil?
       hpxml.building_occupancy.number_of_residents = Geometry.get_occupancy_default_num(nbeds)
       hpxml.building_occupancy.number_of_residents_isdefaulted = true
     end
-    if hpxml.building_occupancy.weekday_fractions.nil? && hpxml.header.schedules_filepath.nil?
+    schedules_file_includes_occupants = false
+    if not schedules_file.nil?
+      if schedules_file.schedules.keys.include?('occupants')
+        schedules_file_includes_occupants = true
+      end
+    end
```

Comment:
Make this a method with a column_name argument so that you don't copy/paste 5 lines of code for every column.

Also, should we convert the column names to constants and update any references?